hopp/simulation/technologies/wind_source.py

```diff
@@ -1,86 +1,96 @@
+from attrs import define, field
 from typing import Optional, Union, Sequence
 
 import PySAM.Windpower as Windpower
 import PySAM.Singleowner as Singleowner
 
+from hopp.simulation.base import BaseClass, BaseModel
 from hopp.simulation.technologies.wind.floris import Floris
 from hopp.simulation.technologies.power_source import PowerSource
+from hopp.simulation.technologies.wind.pysam_wind import PySAMWind
+from hopp.simulation.technologies.wind.pysam_financial import PySAMFinancial
 from hopp.simulation.technologies.sites import SiteInfo
 from hopp.simulation.technologies.layout.wind_layout import WindLayout, WindBoundaryGridParameters
 from hopp.simulation.technologies.dispatch.power_sources.wind_dispatch import WindDispatch
 from hopp.utilities.log import hybrid_logger as logger
 
 
-class WindPlant(PowerSource):
-    _system_model: Union[Windpower.Windpower, Floris]
-    _financial_model: Singleowner.Singleowner
-    _layout: WindLayout
-    _dispatch: WindDispatch
+MODEL_MAP = {
+    "wind_simulation_model": {
+        "pysam": PySAMWind,
+        "floris": Floris,
+    },
+    "wind_financial_model": {
+        "pysam": PySAMFinancial,
+    }
+}
 
-    def __init__(self,
-                 site: SiteInfo,
-                 farm_config: dict,
-                 rating_range_kw: tuple = (1000, 3000),
-                 ):
+@define
+class WindPlant(PowerSource, BaseClass):
+    site: SiteInfo = field()
+    farm_config: dict = field(converter=dict)
+    rating_range_kw: tuple = field(default=(1000, 3000))
+
+    _system_model: Union[PySAMWind, Floris] = field(init=False)
```

Comment:
I've been thinking a bit about the use of leading underscores on these attributes. A couple notes on this:
- `attrs` does not respect them internally: https://www.attrs.org/en/stable/init.html#private-attributes-and-aliases
- They do not really follow the "private" convention anyway, as they are used by other classes within HOPP

I'm experimenting with having these attributes not use underscores on #212

Reply:
I agree, I think we can remove them.